Use r7i.4xlarge for B200 build

zxiiro opened the pull request:
The build system is oversized for what is necessary. Reduce the size to optimize costs. The default workflow runner is `linux.r7i.4xlarge` so we are just removing the runner definition in the workflow so that it uses the default.

nWEIdia reviewed (commented):
Should we just change b200 related jobs, as the title suggests?

zxiiro commented:
@pytorchbot revert -m="This seems to be breaking build when compile is not using sscache. Needs more investigation." -c=nosignal

zxiiro committed:
Use r7i.4xlarge in B200 build jobs

The build instance r5.12xlarge is oversized for what is necessary
to run a successful build job. Reduce the size to optimize costs.

Relates to pytorch/test-infra#7175.

Signed-off-by: Thanh Ha <thanh.ha@linuxfoundation.org>
